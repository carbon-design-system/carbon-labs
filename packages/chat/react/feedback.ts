/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { createComponent } from '@lit/react';
import Feedback from '../components/feedback/feedback.js';

export const C4AIFeedback = createComponent({
  tagName: 'c4ai-feedback',
  elementClass: Feedback,
  react: React,
  events: {},
});
