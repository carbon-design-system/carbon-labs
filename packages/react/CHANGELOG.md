# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 0.1.0 (2025-01-30)


### Bug Fixes

* get web-component test for example-button working ([#357](https://github.com/carbon-design-system/carbon-labs/issues/357)) ([5ae7525](https://github.com/carbon-design-system/carbon-labs/commit/5ae7525867850c2e2bc6fd80834b054759437412))


### Features

* script to generate base component in react + web-components ([#354](https://github.com/carbon-design-system/carbon-labs/issues/354)) ([3c24d37](https://github.com/carbon-design-system/carbon-labs/commit/3c24d375f2b8773f7c4dd704197ae6267d919685))
* **storybook:** add root level storybook instance for landing page ([#358](https://github.com/carbon-design-system/carbon-labs/issues/358)) ([473596b](https://github.com/carbon-design-system/carbon-labs/commit/473596b99b5aef9c1dbe261345d24b3033dd56e2))
* **TextHighlighter:** Add TextHighlighter component - React ([#384](https://github.com/carbon-design-system/carbon-labs/issues/384)) ([61dca6b](https://github.com/carbon-design-system/carbon-labs/commit/61dca6ba71aafcab4a101cb21fc898a630c4c333))
