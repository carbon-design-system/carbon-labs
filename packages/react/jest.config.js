/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

export default {
  testEnvironment: 'jsdom',
  preset: 'ts-jest',
  moduleNameMapper: {
    '\\.(css|less|scss|sass)$': '<rootDir>/__mocks__/styleMock.js',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
  transform: {
    '^.+\\.(ts|tsx)$': 'ts-jest',
    '^.+\\.(js|jsx)$': ['babel-jest', { configFile: './.babelrc' }],
  },
  transformIgnorePatterns: ['/node_modules/(?!@carbon-labs)'],
  setupFilesAfterEnv: ['<rootDir>/jest.setup.ts'],

  // Test file patterns to match
  testMatch: [
    '<rootDir>/src/components/**/__tests__/**/*.[jt]s?(x)',
    '<rootDir>/src/**/?(*.)+(spec|test).[jt]s?(x)',
  ],
};
