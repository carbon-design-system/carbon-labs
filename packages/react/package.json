{
  "name": "@carbon/labs-react",
  "version": "0.0.0",
  "private": true,
  "type": "module",
  "description": "React components for Carbon Labs",
  "license": "Apache-2.0",
  "repository": "https://github.com/carbon-design-system/carbon-labs",
  "scripts": {
    "storybook": "storybook dev -p 3000",
    "storybook:build": "storybook build",
    "test": "jest",
    "test:updateSnapshot": "jest -u"
  },
  "files": [
    "es",
    "lib"
  ],
  "peerDependencies": {
    "react": "^16.8.6 || ^17.0.1 || ^18.2.0",
    "react-dom": "^16.8.6 || ^17.0.1 || ^18.2.0"
  },
  "dependencies": {
    "@carbon/react": "^1.71.1",
    "@carbon/styles": "^1.70.0",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "devDependencies": {
    "@babel/core": "^7.26.0",
    "@babel/plugin-proposal-export-default-from": "^7.25.9",
    "@babel/plugin-transform-class-properties": "^7.25.9",
    "@babel/plugin-transform-export-namespace-from": "^7.25.9",
    "@babel/plugin-transform-react-constant-elements": "^7.25.9",
    "@babel/preset-env": "^7.26.0",
    "@babel/preset-react": "^7.26.3",
    "@babel/preset-typescript": "^7.26.0",
    "@mdx-js/react": "^3.1.0",
    "@rollup/plugin-babel": "^6.0.0",
    "@rollup/plugin-commonjs": "^28.0.0",
    "@rollup/plugin-node-resolve": "^16.0.0",
    "@rollup/plugin-typescript": "^12.1.1",
    "@storybook/addon-docs": "^8.4.7",
    "@storybook/addon-essentials": "^8.4.7",
    "@storybook/addon-interactions": "^8.4.7",
    "@storybook/addon-onboarding": "^8.4.7",
    "@storybook/addon-storysource": "^8.4.7",
    "@storybook/addon-webpack5-compiler-babel": "^3.0.3",
    "@storybook/blocks": "^8.4.7",
    "@storybook/react": "^8.4.7",
    "@storybook/react-webpack5": "^8.4.7",
    "@storybook/test": "^8.4.7",
    "@testing-library/jest-dom": "^6.0.0",
    "@testing-library/react": "^16.1.0",
    "@testing-library/user-event": "^14.5.2",
    "@types/babel__core": "^7",
    "@types/babel__preset-env": "^7",
    "@types/jest": "^29.0.0",
    "@types/node": "^16.18.121",
    "@types/prop-types": "^15",
    "@types/react": "^18.3.12",
    "@types/react-dom": "^18.3.1",
    "autoprefixer": "^10.4.20",
    "babel-jest": "^29.7.0",
    "babel-plugin-dev-expression": "^0.2.3",
    "browserslist-config-carbon": "^11.2.0",
    "css-loader": "^7.1.2",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "mini-css-extract-plugin": "^2.9.2",
    "postcss": "^8.4.49",
    "postcss-loader": "^8.1.1",
    "prop-types": "^15.8.1",
    "remark-gfm": "^4.0.0",
    "rollup-plugin-copy": "^3.5.0",
    "rollup-plugin-strip-banner": "^3.0.0",
    "sass-loader": "^16.0.4",
    "storybook": "^8.4.7",
    "storybook-addon-accessibility-checker": "^3.1.61-rc.3",
    "style-loader": "^4.0.0",
    "ts-jest": "^29.2.5",
    "typescript": "^5.2.2",
    "typescript-config-carbon": "^0.3.0",
    "webpack": "^5.97.0"
  }
}
