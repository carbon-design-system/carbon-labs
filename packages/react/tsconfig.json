{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "types": ["jest", "@testing-library/jest-dom"],
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": ["src", "__tests__/**/*"]
}
