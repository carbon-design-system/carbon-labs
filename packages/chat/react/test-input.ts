/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { createComponent } from '@lit/react';
import TestInput from '../components/test-input/test-input.js';

export const C4AITestInput = createComponent({
  tagName: 'c4ai-test-input',
  elementClass: TestInput,
  react: React,
  events: {},
});
