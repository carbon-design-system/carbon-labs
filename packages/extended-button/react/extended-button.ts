/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { createComponent } from '@lit/react';
import ExtendedButton from '../components/extended-button/extended-button.js';

export const C4AIExtendedButton = createComponent({
  tagName: 'c4ai-extended-button',
  elementClass: ExtendedButton,
  react: React,
  events: {},
});
