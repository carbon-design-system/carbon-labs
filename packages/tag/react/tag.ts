/**
 * @license
 *
 * Copyright IBM Corp. 2024
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { createComponent } from '@lit/react';
import Tag from '../components/tag/tag.js';

export const CLABSTag = createComponent({
  tagName: 'clabs-tag',
  elementClass: Tag,
  react: React,
  events: {},
});
