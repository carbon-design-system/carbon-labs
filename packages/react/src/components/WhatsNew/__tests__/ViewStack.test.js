/**
 * @license
 *
 * Copyright IBM Corp. 2025
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { render, act } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { ViewStack, View } from '../index';

jest.mock('./whats-new.scss', () => ({}));
describe('ViewStack', () => {
  const prefix = 'clabs--whats-new';
  describe('renders as expected - Component API', () => {
    it('should match snapshot', async () => {
      const { container, getByTestId } = render(
        <ViewStack data-testid="TestViewStack">
          <View></View>
        </ViewStack>
      );

      const viewStackElement = getByTestId('TestViewStack');
      expect(container).toMatchSnapshot();
      expect(viewStackElement).toBeInTheDocument();
    });

    it('First element should be active by default', () => {
      const testRef = React.createRef();
      const { getByTestId } = render(
        <ViewStack
          // eslint-disable-next-line react/forbid-component-props
          style={{ blockSize: 100 }}
          ref={testRef}
          data-testid="TestViewStack">
          <View data-testid="TestView1">
            <div>Slide 1</div>
          </View>
          <View data-testid="TestView2">
            <div>Slide 2</div>
          </View>
          <View data-testid="TestView3">
            <div>Slide 3</div>
          </View>
        </ViewStack>
      );
      const view1 = getByTestId('TestView1');
      const view2 = getByTestId('TestView2');
      const view3 = getByTestId('TestView3');

      expect(view1).toHaveClass(`${prefix}__view`);
      expect(view2).toHaveClass(`${prefix}__view`);
      expect(view3).toHaveClass(`${prefix}__view`);

      expect(view1).toHaveClass(`${prefix}__view-active`);
      expect(view2).not.toHaveClass(`${prefix}__view-active`);
      expect(view3).not.toHaveClass(`${prefix}__view-active`);
    });

    it('Should increment the view', () => {
      const testRef = React.createRef();
      const { getByTestId } = render(
        <ViewStack
          // eslint-disable-next-line react/forbid-component-props
          style={{ blockSize: 100 }}
          ref={testRef}
          data-testid="TestViewStack">
          <View data-testid="TestView1">
            <div>Slide 1</div>
          </View>
          <View data-testid="TestView2">
            <div>Slide 2</div>
          </View>
          <View data-testid="TestView3">
            <div>Slide 3</div>
          </View>
        </ViewStack>
      );
      const view1 = getByTestId('TestView1');
      const view2 = getByTestId('TestView2');
      const view3 = getByTestId('TestView3');

      expect(view1).toHaveClass(`${prefix}__view-active`);
      expect(view2).not.toHaveClass(`${prefix}__view-active`);
      expect(view3).not.toHaveClass(`${prefix}__view-active`);

      act(() => {
        testRef.current?.next();
      });

      expect(view1).toHaveClass(`${prefix}__view-in-stack`);
      expect(view2).toHaveClass(`${prefix}__view-active`);
      expect(view3).not.toHaveClass(`${prefix}__view-active`);
    });

    it('Should navigate forwards and backwards', () => {
      const testRef = React.createRef();
      const { getByTestId } = render(
        <ViewStack
          // eslint-disable-next-line react/forbid-component-props
          style={{ blockSize: 100 }}
          ref={testRef}
          data-testid="TestViewStack">
          <View data-testid="TestView1">
            <div>Slide 1</div>
          </View>
          <View data-testid="TestView2">
            <div>Slide 2</div>
          </View>
          <View data-testid="TestView3">
            <div>Slide 3</div>
          </View>
        </ViewStack>
      );
      const view1 = getByTestId('TestView1');
      const view2 = getByTestId('TestView2');
      const view3 = getByTestId('TestView3');

      expect(view1).toHaveClass(`${prefix}__view-active`);
      expect(view2).not.toHaveClass(`${prefix}__view-active`);
      expect(view3).not.toHaveClass(`${prefix}__view-active`);

      act(() => {
        testRef.current?.next();
      });

      expect(view1).toHaveClass(`${prefix}__view-in-stack`);
      expect(view2).toHaveClass(`${prefix}__view-active`);
      expect(view3).not.toHaveClass(`${prefix}__view-active`);

      act(() => {
        testRef.current?.back();
      });

      expect(view1).toHaveClass(`${prefix}__view-active`);
      expect(view2).not.toHaveClass(`${prefix}__view-active`);
      expect(view3).not.toHaveClass(`${prefix}__view-active`);
    });

    it('Should navigate to a specific view index', () => {
      const testRef = React.createRef();
      const { getByTestId } = render(
        <ViewStack
          // eslint-disable-next-line react/forbid-component-props
          style={{ blockSize: 100 }}
          ref={testRef}
          data-testid="TestViewStack">
          <View data-testid="TestView1">
            <div>Slide 1</div>
          </View>
          <View data-testid="TestView2">
            <div>Slide 2</div>
          </View>
          <View data-testid="TestView3">
            <div>Slide 3</div>
          </View>
        </ViewStack>
      );
      const view1 = getByTestId('TestView1');
      const view2 = getByTestId('TestView2');
      const view3 = getByTestId('TestView3');

      expect(view1).toHaveClass(`${prefix}__view-active`);
      expect(view2).not.toHaveClass(`${prefix}__view-active`);
      expect(view3).not.toHaveClass(`${prefix}__view-active`);

      act(() => {
        testRef.current?.pushViewIndex(2);
      });

      expect(view1).toHaveClass(`${prefix}__view-in-stack`);
      expect(view2).toHaveClass(`${prefix}__view`);
      expect(view3).toHaveClass(`${prefix}__view-active`);

      act(() => {
        testRef.current?.pushViewIndex(1);
      });

      expect(view1).toHaveClass(`${prefix}__view-in-stack`);
      expect(view2).toHaveClass(`${prefix}__view-active`);
      expect(view3).toHaveClass(`${prefix}__view-in-stack`);

      act(() => {
        testRef.current?.home();
      });

      expect(view1).toHaveClass(`${prefix}__view-active`);
      expect(view2).not.toHaveClass(`${prefix}__view-active`);
      expect(view2).not.toHaveClass(`${prefix}__view-in-stack`);
      expect(view3).not.toHaveClass(`${prefix}__view-active`);
      expect(view3).not.toHaveClass(`${prefix}__view-in-stack`);
    });
  });
});
