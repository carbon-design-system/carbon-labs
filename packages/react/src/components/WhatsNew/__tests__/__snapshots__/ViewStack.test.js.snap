// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewStack renders as expected - Component API should match snapshot 1`] = `
<div>
  <div
    aria-label="View stack"
    class="clabs--whats-new__view-stack"
    data-testid="TestViewStack"
    role="region"
    tabindex="0"
  >
    <ul>
      <li
        class="clabs--whats-new__view clabs--whats-new__view-active"
        data-index="0"
      />
    </ul>
    <div
      aria-atomic="true"
      aria-live="polite"
      class="clabs--whats-new__view-stack-announcer"
    >
      Displaying view 1 of 1
    </div>
  </div>
</div>
`;
